Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -273,40 +273,37 @@ public class ByteBuffer: Codable {
     }
 }
 
-extension ByteBuffer: AwsStream {
-    public var status: aws_stream_status {
-        return aws_stream_status(is_end_of_stream: self.currentIndex == array.count, is_valid: true)
-    }
+extension ByteBuffer: IStreamable {
 
-    public var length: UInt {
-        return UInt(array.count)
+    public func length() -> UInt64 {
+        return UInt64(array.count)
     }
 
-    public func seek(offset: Int64, basis: aws_stream_seek_basis) -> Bool {
-        let targetOffset: Int64
-        if basis.rawValue == AWS_SSB_BEGIN.rawValue {
-            targetOffset = offset
+    public func seek(offset: Int64, streamSeekType: StreamSeekType) throws {
```

Comment:
Nit: Doc comments for this (and all public methods and properties) would be awesome.

Reply:
In this particular case, I have added the documentation in the protocol for Seek & Read functions.